contracts/message/apps/RFQ.sol

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "../framework/MessageSenderApp.sol";
+import "../framework/MessageReceiverApp.sol";
+import "../../safeguard/Pauser.sol";
+
+/** @title rfq contract */
+contract RFQ is MessageSenderApp, MessageReceiverApp, Pauser, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    struct AgreementDetail {
+        uint64 srcChainId;
+        address srcToken;
+        uint256 amount1;
+        uint64 dstChainId;
+        address dstToken;
+        uint256 amount2;
+        uint64 releaseDeadline;
+        uint64 nonce;
+        address usrAddr;
+        address dstRecipient;
+        address refundTo;
+        address srcRecipient;
+    }
+
+    mapping(uint64 => address) public apps;
+    mapping(bytes32 => bool) public unconsumedMsg;
+    mapping(bytes32 => bool) public orderBooks;
+    mapping(bytes32 => bool) public filledOrder;
+    uint32 public feePercGlobal;
+    // dstChainId => feePercOverride
+    mapping(uint64 => uint32) public feePercOverride;
+    uint64 public safeTime;
+
+    event Msg1Sent(bytes32 hash, AgreementDetail detail, address srcRecipient, uint64 submissionDeadline);
+    event Msg2Sent(bytes32 hash);
+    event Msg3Sent(bytes32 hash);
+    event MessageReceived(bytes32 hash);
+    event Msg2Executed(bytes32 hash, address srcRecipient, address srcToken, uint256 amount);
+    event Msg3Executed(bytes32 hash, address refundTo, address srcToken, uint256 amount);
+    event AppUpdated(uint64 chainId, address app);
+    event FeePercUpdated(uint64 chainId, uint32 feePerc);
+    event SafeTimeUpdated(uint64 safeTime);
+
+    constructor(address _messageBus) {
+        messageBus = _messageBus;
+    }
+
+    function sendMessage1(AgreementDetail calldata _agrDetail, uint64 _submissionDeadline)
```

Comment:
let function return the quote hash, could be helpful for contract sender integration

Reply:
right